pkg/docker/fast_push.go

```diff
@@ -128,12 +145,30 @@ func startUploadURL(objectType string, digest string) url.URL {
 	return uploadUrl
 }
 
+func verificationURL(objectType string, digest string, uuid string) url.URL {
+	verificationUrl := baseURL()
+	verificationUrl.Path = strings.Join([]string{"", "uploads", objectType, "sha256", digest, uuid, "verification"}, "/")
+	return verificationUrl
+}
+
+func checkVerificationStatus(req *http.Request, client *http.Client) (bool, error) {
+	checkResp, err := client.Do(req)
+	if err != nil {
+		return false, err
+	}
+	defer checkResp.Body.Close()
+	return checkResp.StatusCode == http.StatusOK, nil
+}
+
 func uploadFile(objectType string, digest string, path string, token string) error {
 	console.Debug("uploading file: " + path)
 
 	uploadUrl := startUploadURL(objectType, digest)
 	client := &http.Client{}
-	req, _ := http.NewRequest("POST", uploadUrl.String(), nil)
+	req, err := http.NewRequest("POST", uploadUrl.String(), nil)
```

Comment:
I don't see a `context.Context` available, but getting one passed down so that cancellation and cleanup are handled correctly will be real nice 😁 and so that you can use `http.NewRequestWithContext`.